deploy/operator-role-olm.yaml

```diff
@@ -0,0 +1,131 @@
+# ---------------------------------------------------------------------------
```

Comment:
I think the main reason why we have a double version of roles in Camel K is that you could install camel k in a namespace as a standard user, and the core operator is installed, even if you don't have rights to see some resources (e.g. you don't have access to Knative resources for some reasons).

So e.g. this file usually contain everything and cannot be installed by standard users, so we use it only in OLM where there are no restrictions. And we have other files that contain all the various parts, that the installer tries to install one at time and doesn't fail in case of error.

You can decide to manage things the same way, or just say "you need to be a cluster admin to install yaks", to simplify things.

Reply:
Yep I also noticed that openshift and kubernetes roles in YAKS are identical at the moment. So et the end everything could be consolidated to one single operator-role configuration. But I think I leave it as it is now so we are set for possible future development where YAKS roles for openshift and kubernetes start diverging.